server/src/main/java/org/apache/iotdb/db/metadata/MLogTxtWriter.java

```diff
@@ -166,6 +166,14 @@ public void setStorageGroup(String storageGroup) throws IOException {
     lineNumber.incrementAndGet();
   }
 
+  public void updateStorageGroup(String storageGroup) throws IOException {
+    String outputStr =
+        MetadataOperationType.UPDATE_STORAGE_GROUP + "," + storageGroup + LINE_SEPARATOR;
+    ByteBuffer buff = ByteBuffer.wrap(outputStr.getBytes());
+    channel.write(buff);
+    lineNumber.incrementAndGet();
+  }
+
```

Comment:
Is it possible to recover the versions of an SG after a restart?

Reply:
Sure, we have serialized the versions and after a restart, the plan can be deserialized correctly, the above codes is just one parse util that parses mlog.bin to mlog.txt